### PR TITLE
[lab2] add code documentation :memo:

### PR DESCRIPTION
**Pull request by afikbar**

### CODE REVIEW

File: lab2/src/FeatureVector.cpp [hunk 1]
```diff
@@ -2,10 +2,21 @@
 #include "include/FeatureVector.h"
 #include "include/Statistics.h"
 
+/**
+ * Constructor
+ * construct a FeatureVector using Feature Name only.
+ * @param name string
```
Comment (odedlaz) [inline lab2/src/FeatureVector.cpp]: Add an explanation for what *name* does, for example:
```cpp
/**
* Constructor
* construct a FeatureVector using Feature Name only.
* @param name string The name of the variable (aka: FeatureVector)
```

File: lab2/src/FeatureVector.cpp [hunk 2]
```diff
@@ -2,10 +2,21 @@
...
+ */
 FeatureVector::FeatureVector(std::string name) : _name(name) {
     _features = std::vector<double>();
 }
 
+/**
+ * Constructor
+ * construct a FeatureVector using name and vector of data.
+ * @param name string
+ * @param features vector<double>
```
Comment (odedlaz) [inline lab2/src/FeatureVector.cpp]: Same as above :arrow_up:

File: lab2/src/Statistics.cpp [hunk 3]
```diff
@@ -44,10 +62,23 @@ double statistics::variance(const std::vector<double> &xv) {
     return sum / xv.size();
 }
 
+/**
+ * Standard Deviation
+ * calculates the standard deviation of given vector
+ * @param xv Type vector<double>
```
Comment (odedlaz) [inline lab2/src/Statistics.cpp]: Same as above :arrow_up:

File: lab2/src/include/Statistics.h [hunk 4]
```diff
@@ -3,7 +3,10 @@
 
 #include <cstdlib>
 #include <vector>
-
+/**
+ * Statistics
+ * handles statistics calculations for vector<double> types.
+ */
```
Comment (odedlaz) [inline lab2/src/include/Statistics.h]: maybe we need to add documentation for all the functions in the statistics namespace?

File: lab2/src/main.cpp [hunk 5]
```diff
@@ -70,10 +74,10 @@ int main(int argc, char *argv[]) {
         return 1;
     }
 
-    double min = 1.0;
-    double max = 0.0;
-    // create seed correlation for min and max values
+    double min = 1.0;//starting from strongest correlation possible
```
Comment (odedlaz) [inline lab2/src/main.cpp]: Add a space between `;` and `//`. Did you run `Ctrl+Alt+L`?

File: lab2/src/include/Statistics.h (hunk 4 above)
Comment (afikbar) [inline lab2/src/include/Statistics.h]: you mean to explain why they are in namespace?

Comment (odedlaz) [inline lab2/src/include/Statistics.h]: No, I meant that maybe we need to add documentation to what mean / variance / etc.. does.

File: lab2/src/main.cpp (hunk 5 above)
Comment (afikbar) [inline lab2/src/main.cpp]: yes.. i ran again, apparently it doesn't do it automatically

**Pull request merged**